File: backend/app/schemas/restaurant/restaurant.py
from pydantic import BaseModel, ConfigDict


class RestaurantTableResponse(BaseModel):
    """Schema for restaurant table in API responses"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    restaurant_id: int
    seating_count: int


class RestaurantResponse(BaseModel):
    """Schema for restaurant in API responses"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    total_seating: int


class RestaurantDetailResponse(RestaurantResponse):
    """Schema for detailed restaurant view including tables"""

    tables: list[RestaurantTableResponse]

File: frontend/src/types/restaurant.ts
export interface RestaurantTable {
  id: number;
  restaurant_id: number;
  seating_count: number;
}

export interface Restaurant {
  id: number;
  name: string;
  total_seating: number;
  image_url: string | null;
}

export interface RestaurantDetail extends Restaurant {
  tables: RestaurantTable[];
}
